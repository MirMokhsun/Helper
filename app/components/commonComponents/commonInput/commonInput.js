import React, { PureComponent } from 'react';
import { View, TextInput } from 'react-native';
import { styles } from './styles';
import WarningSVG from './svg/warning';
import { WarningInputText } from './commonInputElements/warningInputText';
import { validationTest } from '../../../api/regexFilters/validationInput';

export class CommonInput extends PureComponent {
    constructor(props) {
        super(props)
        this.state = {
            isShowWarning: false,
        }
    }

    checkValue = (text) => {
        const { filter, action } = this.props;
        const isValidValue = validationTest(text, filter);
        if (isValidValue) {
            this.setState({
                isShowWarning: false,
            });
        }
        action({ isValid: isValidValue, value: text.trim() });
    }

    onBlur = () => {
        const { isValid } = this.props;
        if (!isValid) {
            this.setState({
                isShowWarning: true,
            });
        }
    }

    render() {
        const { currentLanguageName, placeholder, errorText, testID, textErrorID, keyboardType, maxWidth, value } = this.props;
        const { isShowWarning } = this.state;
        return (
            <View style={[styles.container, { width: maxWidth || 290 }]}>
                <View style={[styles.textinputwrapper, value ? null : styles.borderGray, isShowWarning ? styles.warningStyle : null,]}>
                    <TextInput
                        keyboardType={keyboardType || 'default'}
                        testID={testID}
                        accessibilityLabel={testID}
                        value={value}
                        onBlur={this.onBlur}
                        selectionColor="#5A6772"
                        placeholderTextColor='#9FA0B2'
                        underlineColorAndroid='transparent'
                        onChangeText={this.checkValue}
                        style={[styles.textinput, currentLanguageName === 'ar' ? styles.textinputWithTextAlightRight : null]}
                        placeholder={placeholder}
                        autoCorrect={false} />
                    <WarningSVG isShowWarning={isShowWarning} />
                </View>
                <WarningInputText maxWidth={maxWidth} isShowWarningText={isShowWarning} text={errorText} testID={textErrorID} />
            </View>
        )
    }
}