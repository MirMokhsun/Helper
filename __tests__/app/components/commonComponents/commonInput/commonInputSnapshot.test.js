import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { CommonInput } from '../../../../../app/components/commonComponents/commonInput/commonInput';
import renderer from 'react-test-renderer';

describe('CommonInput', () => {
    it('create snapshot CommonInput', () => {
        const render = new ShallowRenderer();
        let commonInputSnapshot = render.render(<CommonInput
            errorText={'errorFirstName'}
            filter={/$/}
            action={() => { }}
            isValid={true}
        />);
        expect(commonInputSnapshot).toMatchSnapshot();
    });
    it('test function onBlur', () => {
        let commonInput = renderer.create(<CommonInput isValid={false} currentLanguageName={'ar'} />);
        commonInput.getInstance().onBlur();
        expect(commonInput.getInstance().state.isShowWarning).toEqual(true);
        expect(commonInput.toJSON()).toMatchSnapshot();

        let commonInputIsValid = renderer.create(<CommonInput isValid={true} />).getInstance();
        commonInputIsValid.onBlur();
        expect(commonInputIsValid.state.isShowWarning).toEqual(false);
    });
    it('test function checkValue', () => {
        let commonInput = renderer.create(<CommonInput
            errorText={'errorFirstName'}
            filter={/.{6,}/}
            action={() => { }}
            isValid={false}
            currentLanguageName={'ar'}
        />);
        commonInput.getInstance().checkValue('some');
        commonInput.getInstance().checkValue('some-text');
        expect(commonInput.getInstance().state.isShowWarning).toEqual(false);
        expect(commonInput.toJSON()).toMatchSnapshot();
    });
});