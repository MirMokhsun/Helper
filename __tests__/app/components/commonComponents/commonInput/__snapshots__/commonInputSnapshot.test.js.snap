// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonInput create snapshot CommonInput 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 50,
      },
      Object {
        "width": 290,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomWidth": 1,
          "borderColor": "#1459D2",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "space-between",
          "paddingRight": 5,
          "width": "100%",
        },
        Object {
          "borderColor": "#D3D9E7",
        },
        null,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholderTextColor="#9FA0B2"
      rejectResponderTermination={true}
      selectionColor="#5A6772"
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "color": "#1459D2",
            "flex": 1,
            "fontFamily": "Roboto-Light",
            "fontSize": 18,
            "height": "100%",
            "marginRight": 10,
            "padding": 0,
          },
          null,
        ]
      }
      underlineColorAndroid="transparent"
    />
    <WarningSVG
      isShowWarning={false}
    />
  </View>
  <WarningInputText
    isShowWarningText={false}
    text="errorFirstName"
  />
</View>
`;

exports[`CommonInput test function checkValue 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 50,
      },
      Object {
        "width": 290,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomWidth": 1,
          "borderColor": "#1459D2",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "space-between",
          "paddingRight": 5,
          "width": "100%",
        },
        Object {
          "borderColor": "#D3D9E7",
        },
        null,
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholderTextColor="#9FA0B2"
      rejectResponderTermination={true}
      selectionColor="#5A6772"
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "color": "#1459D2",
            "flex": 1,
            "fontFamily": "Roboto-Light",
            "fontSize": 18,
            "height": "100%",
            "marginRight": 10,
            "padding": 0,
          },
          Object {
            "textAlign": "right",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
</View>
`;

exports[`CommonInput test function onBlur 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "column",
        "height": 50,
      },
      Object {
        "width": 290,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomWidth": 1,
          "borderColor": "#1459D2",
          "flexDirection": "row",
          "height": 35,
          "justifyContent": "space-between",
          "paddingRight": 5,
          "width": "100%",
        },
        Object {
          "borderColor": "#D3D9E7",
        },
        Object {
          "borderColor": "#F86F6F",
        },
      ]
    }
  >
    <TextInput
      allowFontScaling={true}
      autoCorrect={false}
      keyboardType="default"
      onBlur={[Function]}
      onChangeText={[Function]}
      placeholderTextColor="#9FA0B2"
      rejectResponderTermination={true}
      selectionColor="#5A6772"
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "color": "#1459D2",
            "flex": 1,
            "fontFamily": "Roboto-Light",
            "fontSize": 18,
            "height": "100%",
            "marginRight": 10,
            "padding": 0,
          },
          Object {
            "textAlign": "right",
          },
        ]
      }
      underlineColorAndroid="transparent"
    />
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={20}
      bbWidth={20}
      height={20}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          undefined,
          null,
          Object {
            "flex": 0,
            "height": 20,
            "width": 20,
          },
        ]
      }
      vbHeight={20}
      vbWidth={20}
      width={20}
    >
      <RNSVGGroup
        fill={
          Array [
            0,
            4278190080,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        font={Object {}}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={Array []}
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      >
        <RNSVGPath
          d="M10 0C4.48 0 0 4.48 0 10s4.48 10 10 10 10-4.48 10-10S15.52 0 10 0zm1 15H9v-2h2v2zm0-4H9V5h2v6z"
          fill={
            Array [
              0,
              4292501600,
            ]
          }
          fillOpacity={1}
          fillRule={1}
          matrix={
            Array [
              1,
              0,
              0,
              1,
              0,
              0,
            ]
          }
          opacity={1}
          originX={0}
          originY={0}
          propList={
            Array [
              "fill",
            ]
          }
          rotation={0}
          scaleX={1}
          scaleY={1}
          skewX={0}
          skewY={0}
          stroke={null}
          strokeDasharray={null}
          strokeDashoffset={null}
          strokeLinecap={0}
          strokeLinejoin={0}
          strokeMiterlimit={4}
          strokeOpacity={1}
          strokeWidth={1}
          vectorEffect={0}
          x={0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={
      Array [
        Object {
          "height": 15,
          "justifyContent": "flex-start",
        },
        Object {
          "width": 290,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#F86F6F",
          "fontFamily": "Roboto-Light",
          "fontSize": 11,
          "lineHeight": 15,
          "textAlign": "right",
        }
      }
    />
  </View>
</View>
`;
